internal/controller/drcluster_controller.go

```diff
@@ -428,13 +423,27 @@ func (r DRClusterReconciler) processCreateOrUpdate(u *drclusterInstance) (ctrl.R
 			u.validatedSetFalseAndUpdate("DrClustersDeployStatusCheckFailed", err))
 	}
 
+	if err := u.ensureDRClusterConfig(); err != nil {
+		return ctrl.Result{}, fmt.Errorf(
+			"failed to ensure DRClusterConfig: %w",
+			u.validatedSetFalseAndUpdate("DRClusterConfigInProgress", err),
+		)
+	}
+
 	setDRClusterValidatedCondition(&u.object.Status.Conditions, u.object.Generation, "Validated the cluster")
 
+	err = u.clusterMModeHandler()
+	if err != nil {
+		requeue = true
+
+		u.log.Info("Error during processing maintenance modes", "error", err)
+	}
+
 	if err := u.statusUpdate(); err != nil {
 		u.log.Info("failed to update status", "failure", err)
 	}
 
-	return ctrl.Result{Requeue: requeue || u.requeue}, reconcileError
+	return ctrl.Result{Requeue: requeue || u.requeue}, nil
```

Comment:
If there is an error during fencing - we will not return error?

Reply:
If there is an error in fencing we will note that in the status and log it, and we set requeue to true (for future reconciles). Returning an error is not required at this point.